app/uk/gov/hmrc/tai/viewModels/WhatDoYouWantToDoViewModel.scala

```diff
@@ -16,49 +16,9 @@
 
 package uk.gov.hmrc.tai.viewModels
 
-import uk.gov.hmrc.tai.model.domain.TaxCodeMismatch
-import uk.gov.hmrc.tai.util.MapForGoogleAnalytics
-import uk.gov.hmrc.tai.util.constants.GoogleAnalyticsConstants
-
-import scala.collection.immutable.ListMap
+import java.time.LocalDate
 
 case class WhatDoYouWantToDoViewModel(
   isCyPlusOneEnabled: Boolean,
-  hasTaxCodeChanged: Boolean = false,
   showJrsTile: Boolean,
-  taxCodeMismatch: Option[TaxCodeMismatch] = None) {
-
-  def showTaxCodeChangeTile(): Boolean =
```

Comment:
just out of curiosity, was this moved to the controller as this is best practice?

Reply:
i used the case class a placeholder for data and nothing else. In such case to me the only important thing is to hold the maybeMostRecentTaxCodeChangeDate. As this is the model and must have the date now, I could have used the apply/factory method to generate the date but that also meant knowing an optional date that comes from a service call... I thought that would be overcomplicating things that should be kept as part of the controller (the controller is the one that interacts with service calls)

finally, that shouldn't have been a def as the attributes used are values that will never change...